src/OpenSage.Game/Data/InstallationLocator.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace OpenSage.Data
+{
+    public enum GameId
+    {
+        Generals,
+        ZeroHour
+    }
+
+    public struct GameInstallation
+    {
+        public readonly GameId Game;
+        public readonly string Path;
+
+        public GameInstallation(GameId game, string path)
+        {
+            Game = game;
+            Path = path;
+        }
+
+        public string DisplayName => GetDisplayName(Game);
+
+        private static string GetDisplayName(GameId game)
+        {
+            switch (game)
+            {
+                case GameId.Generals: return "C&C Generals";
+                case GameId.ZeroHour: return "C&C Generals Zero Hour";
+            }
+
+            throw new ArgumentException($"{game} is not supported.", nameof(game));
+        }
+    }
+
+    public static class Games
+    {
+        public static IEnumerable<GameId> GetAll()
+        {
+            yield return GameId.Generals;
+            yield return GameId.ZeroHour;
+        }
+    }
+
+    public interface IInstallationLocator
+    {
+        IEnumerable<GameInstallation> FindInstallations(GameId game);
+    }
+
+    public class RegistryInstallationLocator : IInstallationLocator
+    {
+        private static readonly (string, string)[] GeneralsKeys = new[] { (@"SOFTWARE\Electronic Arts\EA Games\Generals", "InstallPath") };
```

Comment:
These are arrays, because games could have different editions / releases with different registry keys. If that turns out not to be the case, we can just store single keys.